project_weather_app/main.py

```diff
@@ -0,0 +1,41 @@
+from database import WeatherDatabase
+from data_handler import DataHandler
+from data_service import DataServiceFactory
+from location import Location
+
+MODE = "API" # data retrieval mode: ["API"|"MOCK"]
```

Comment:
This is fine, though once your application grows, the keyword "API" is no longer descriptive enough to cover your full application. And you may want to only "mock" specific parts while others should use the actual components (I.e. use actual database BUT mocked API), so you could think about controlling this in a more sophisticated way